### DISCUSSION TITLE
Breaking change after PR #4572

### DISCUSSION THREAD

**Discussion by cvuillemez** [rabbitmq/rabbitmq-server#10601] [Other] [answered]
### Describe the bug

Seems the [PR #4572](https://github.com/rabbitmq/rabbitmq-server/pull/4572) is a breaking change:
queue declaration without argument can fail (after upgrade from 3.9.x to 3.12.x).

### Reproduction steps

1. Declare a classic queue with argument:
```
arguments = { "x-queue-master-locator": "min-masters" }
channel.queue_declare(queue='test', durable=True, arguments=arguments)
```

2. Then declare again the (existing) queue without argument:
```
channel.queue_declare(queue='test', durable=True)
[...]
pika.exceptions.ChannelClosedByBroker: (406, "PRECONDITION_FAILED - inequivalent arg 'x-queue-master-locator' for queue 'test' in vhost 'test': received none but current is the value 'min-masters' of type 'longstr'")
```

### Expected behavior

Do not throw an exception, just ignore the missing argument (?)

**Comment by michaelklishin** [comment, marked as answer]
@cvuillemez we have no plans to further tweak x-argument equivalence checks. Client-facing changes are relatively rare, and the protocol changes happen every few years.

Backwards compatibility with [release series that have reached EOL](https://www.rabbitmq.com/docs/versions) is not and has never been our goal. We test upgrades from the previous (N-1) release series, sometimes N-2. 3.9 to 3.12 is N-3.
